Decisions/scs-XXXX-v1-requirements-for-container-registry.md

```diff
@@ -0,0 +1,362 @@
+---
+title: Requirements for container registry
+type: Decision Record
+status: Draft
+track: KaaS
+---
+
+# Introduction  
+
+A container registry is a repository or collection of repositories that store and
+access container images. It acts as the intermediary for sharing container images
+between systems and could be connected directly to container orchestration platforms
+like Kubernetes.
+
+Container registries could be publicly accessible e.g. Docker Hub, could be
+self-hosted or hosted by cloud service providers (CSP). These container registries may
+apply various access control mechanisms to restrict public access and make them private.
+Both solutions offer a wide range of features that may or may not attract potential
+users and CSPs.
+
+# Motivation
+
+This proposal is motivated by use cases in which CSPs would like to offer SCP-compliant
+private container registries to their customers. The specific use cases should be
+discussed, but overall CSP could offer a private container registry as a service or
+CSP could offer a recipe for customers to deploy the private registry themselves
+utilizing CSP infrastructure. In both cases, the private container registry should
+be SCS-compliant and should fulfill a set of needed requirements e.g. for security and
+privacy.
+
+The idea and purpose of this document is to specify what requirements a
+specific technical container registry implementation (i.e. software solution) needs to 
+fulfill in the context of SCS.
+
+Another purpose is the selection of an appropriate container registry
+implementation that meets all defined requirements to make architectural
+decision on what implementation is fully SCS-compliant and will be used by the SCS.
```

Comment:
Having read this again: It's unclear to me what this sentence is saying in regard to "SCS compliance".

Does it mean that a container registry is SCS compliant, if ...?

1. it is feature complete according to this record (suggested by the wording), OR
2. has exactly the feature set and interfaces of the solution which is chosen in this record, OR
3. it offers a baseline of *standardized* features defined explicitly defined as requirements of "SCS compliance" (which AFAIK is not done yet)

If the first or second option is selected, I think it would be worth writing that down more explicitly.

OTOH, if this should not be the topic of this record, mentions of "SCS compliance" probably should be omitted or be prepended with a disclaimer like "yet to be defined SCS compliance".

Reply:
This is a truly important point. The quoted sentence seems to imply some kind of normative nature of the document that might not have been intended.

Reply:
Thanks for pointing this out! You are right, it was not intended to define what "SCS compliance"  is.

This decision record just wants to pick one container registry implementation from available open-source solutions. 

I assumed here, that the selected container registry (Harbor in this case) will be SCS-compliant based on this document because it meets all requirements defined by this document (decision record). So the first option you mentioned. 

I would like to suggest writing down a footnote to explain what  "SCS compliance" means in the context of this decision record, e.g.:

```
SCS-compliant[^1]

[^1]: It is not intended to define what in general SCS compliance means. This decision record assumes that the selected container registry will be SCS-compliant based on this decision record because it passed all defined checks and meets all defined requirements.
```

Would it be sufficient?

Reply:
I am not sure what mentioning `SCS-compliant` adds in addition to saying that the selected tool (i.e. Harbor) fulfills all the requirements laid out in the document.

Reply:
@wwentland I just wanted to explain the usage of "SCS-compliance" here as was suggested by @joshmue

The other option which was suggested by @joshmue was to remove SCS compliance from the document or mention it with a disclaimer like "yet to be defined SCS compliance" 

IMO I would like to avoid these because IMO this document is an SCS document written by the SCS community, therefore would be great to discuss and resolve the issue of "what does SCS compliance mean ". 
In addition, I [found](https://github.com/SovereignCloudStack/standards/search?q=SCS+compliant&type=issues) some other documents in the `/standards` repository where these words are used e.g. :
- SCS compliant cloud
- SCS compliant clusters
- scs compliant providers 
- SCS compliant policy repository 


@wwentland @joshmue  as I'm a newbie in the SCS ecosystem, would be great if you could provide a suggestion on what you prefer here (remove SCS compliance, use a disclaimer, reword the explanation ...)

Reply:
>  In addition, I found some other documents in the /standards repository where these words are used e.g. :

I do not think that there is a single file which sums up all SCS standards applicable to e. g. clusters yet. Generally, I think, if something "is compliant" with all requirements in `/Standards` (and `/Decisions`), it is going to be SCS compliant. IIRC it is intended to make this verifiable using conformance checks. Of course, the list of standards and checks still needs to grow.

> IMO I would like to avoid these because IMO this document is an SCS document written by the SCS community, therefore would be great to discuss and resolve the issue of "what does SCS compliance mean ".

Indeed, this is a great topic for a standard [^1] - perhaps just as a distinct one defining SCS compliance explicitly. As-is, personally, I'd drop occurrences of this term altogether and start discussion about it in another PR - which very well could conclude, that Harbor's features and interfaces should also be SCS "compliance" standard.
Maybe @jschoone, @garloff or @fkr also have an opinion about it - otherwise it may be discussed in the next team call.

[^1]: ...or decision. Maybe the types of documents should be split more rigorously in the future. I can see some decision records which actually look like standards.

Reply:
@joshmue thank you for a clear suggestion. I agree that this is an excellent topic for a wider community discussion in the next team call and/or a separate issue.  Thus, the term 'SCS compliance" has been dropped from the document by a separate commit aa70de1f8b9277f9b72b49fce5d4e6c5345ce9ee

Reply:
Added topic to container team meeting agenda.

Reply:
> Indeed, this is a great topic for a standard [1](#user-content-fn-1-9a85709a8e633f9d651716aa5ded36b0) - perhaps just as a distinct one defining SCS compliance explicitly. As-is, personally, I'd drop occurrences of this term altogether and start discussion about it in another PR - which very well could conclude, that Harbor's features and interfaces should also be SCS "compliance" standard. Maybe @jschoone, @garloff or @fkr also have an opinion about it - otherwise it may be discussed in the next team call.
> ## Footnotes
> 
>     1. ...or decision. Maybe the types of documents should be split more rigorously in the future. I can see some decision records which actually look like standards. [leftwards_arrow_with_hook](#user-content-fnref-1-9a85709a8e633f9d651716aa5ded36b0)

Fair footnote. I had the same observation and wondered whether we need to merge ADRs and Standards ...
In any case, I have kept the overall numbers unique, so we should be fine if we need to move docs between the directories. Cleanup patches that move files around are welcome!

To be SCS-compatible, you will need to fulfill all standards that are listed in the scs-compatible.yaml file.
This is relatively new and still small, but I expect it to grow quickly.
While I expect the documents in the Standards directory (and maybe some in the Decisions directory) to become relevant for achieving SCS compliance, this needs an explicit decision by referencing the standard (and hopefully associated compliance tests) to the scs-compatible.yaml file.

Reply:
> Cleanup patches that move files around are welcome!

Opened issue #233 and assigned it to myself to not lose track.

> Added topic to container team meeting agenda.

Result of team call: Discuss topic of  in the "SIG Standardization/Certification". Opened #234.

Resolving thread.